src/main.cpp

```diff
@@ -416,16 +416,16 @@ void check_opencv_formats()
 		if (strings.size() >= 2)
 		{
 			// Portable Network Graphics
-			if ((strings[0] == "PNG") && (strings[1] != "NO"))
+			if ((strings[0] == "PNG"))
 			{
-				opencv_formats["PNG"] = true;
+				opencv_formats["PNG"] = strings[1] != "NO";
 			}
 			// JPEG Files
-			else if ((strings[0] == "JPEG") && (strings[1] != "NO"))
+			else if ((strings[0] == "JPEG"))
 			{
-				opencv_formats["JPEG"] = true;
-				opencv_formats["JPG"] = true;
-				opencv_formats["JPE"] = true;
+				opencv_formats["JPEG"] = (strings[1] != "NO");
+				opencv_formats["JPG"] = (strings[1] != "NO");
+				opencv_formats["JPE"] = (strings[1] != "NO");
```

Comment:
You are right about that. I just thought it was more important to keep the `!= "NO"` checks instead of changing these ones to `== "NO"` or `== "YES"`  (Unsure if this works).
If you rather, I change the if condition to `((strings[0] == "JPEG") && (strings[1] == "NO"))` and then the `opencv_formats` lines to `... = false`, I can do it no problem.

Reply:
You can use =="NO" because getBuildInformation() returns NO or YES or Path of lib

actually use should not use =="YES"

check output sample of getBuildInformation()

https://www.learnopencv.com/get-opencv-build-information-getbuildinformation/

Reply:
About that, I need to rewrite the buildinfo part anyway, to use:
```
#include <opencv2/opencv.hpp>
#include <iostream>
int main() {
	std::cout << cvHaveImageWriter(".png") << "\n";
	std::cout << cvHaveImageWriter(".mp4") << "\n";
}
```

Reply:
Oh,

OK then I just Approve PR if you will make codes again.